RELEASE.md

```diff
@@ -0,0 +1,152 @@
+# Release
+
+The Prometheus Go client library follows a release process similar to the [Prometheus server](https://github.com/prometheus/prometheus/blob/main/RELEASE.md).
```

Comment:
Is it similar? I have the impression that we do releases when we have a feeling it's needed, there's no schedule here 🤔

Reply:
I can explicitly state that we don't have a cadence.